packages/x-tree-view/src/hooks/useTreeItem2Utils/useTreeItem2Utils.tsx

```diff
@@ -112,10 +124,48 @@ export const useTreeItem2Utils = ({
     }
   };
 
+  const toggleItemEditing = () => {
+    if (!hasPlugin(instance, useTreeViewLabel)) {
```

Comment:
Nitpick, in `toggleItemEditing` you do an early return for `!hasPlugin` and in `handleSaveItemLabel` / `handleCancelItemLabelEditing` you include `hasPlugin` in the main if.